lab-yunjoo/lib/note-crud.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const Note = require('../model/note');
+const AppErr = require('../lib/app-error');
+const debug = require('debug')('rito:note-crud');
+
+exports.createNote = function(reqBody){
+  return new Promise((resolve, reject) =>{
+    if(!reqBody.content)
+      return reject(AppErr.error400('note require\'s content'));
+    if(!reqBody.name)
+      return reject(AppErr.error400('note require\'s name'));
+
+    reqBody.timestamp = new Date();
+    const note = new Note(reqBody);
+    note.save() //save on 'note' objects, not 'Note' constructor
+    .then(resolve)
+    .catch(reject);
+  });
+};
+
+exports.fetchNoteByNoteId = function(id){
+  debug('fetchNote');
+  return new Promise((resolve, reject) =>{
+    Note.findOne({_id: id})
+    .then(resolve)
```

Comment:
This works for the most part, but I think that there are situations in which it fails. Your tests work because it will throw an error when it tries to convert '1234' to a valid mongoose id. However, I think that if you were to supply a valid id for a note that had been deleted, this would not throw an error, it would just resolve with null.